### DISCUSSION TITLE
Seeking Help - Single Agent Asking Multiple Questions

### DISCUSSION THREAD

**Discussion by LedonLenzly** [microsoft/autogen#3993] [Q&A]
I would like my agent to ask a series of 6 to 10 consecutive questions of the user. I am not sure where to list those questions (i.e. in the system_message variable of the agent or the chat.append section), Also, I am not sure if additional Python script is needed to create a loop. Can someone point me in the right direction?

ex. Do I declare the questions in this way

agent1 = ConversableAgent (
name= "agent1",
system_message = "You are a person who asks a bunch of questions.
   You are only to ask the questions that are listed below.
            "What is the purpose?", 
            "What will be the location?",
            "What are the size requirements?",
            "What will the content be?", 
            "What is the desired color?",
llm_config=llm_config,
human_input_mode= "NEVER",
is_termination_msg = lambda msg: "terminate" in msg.get("content").lower()  
)

or should it be done in the append chat section?

chats.append(
    {
        "sender": CSRRequirementsAgent,
        "recipient": customer_proxy_agent,
        "message": 
            "Thank you for providing your contact information."
            "Now I would like to learn more about the type of sign that you would like us to build for you.",
        "summary_method": "reflection_with_llm",
        "summary_args": {
        "summary_prompt" : "Return the purpose, location, size, content, color"
                             "into a JSON object only: "
                             "{'purpose': '', 'location': '','size': '','content': '', 'color': ''}",
        },
        "clear_history" : False
    }
)
 
If both approaches are wrong, can you provide a snippet that I can use as an example?

**Comment by ekzhu** [comment]
Sounds like something can benefit from using either stateflow or sequential chat:

1. https://microsoft.github.io/autogen/0.2/docs/topics/groupchat/customized_speaker_selection
2. https://microsoft.github.io/autogen/0.2/docs/tutorial/conversation-patterns#sequential-chats

BTW we will have updated APIs for this in the v0.4 soon.

**Comment by LedonLenzly** [comment]
This is great information. Let me give it a try. Thank you for your response.

Ledon

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
